src/main/resources/templates/registration.html

```diff
@@ -110,6 +117,10 @@ <h1 th:text="#{label.form.title}">form</h1>
         {
             window.location.href = serverContext + "emailError.html";
         }
+        else if(data.responseJSON.error == "InvalidReCaptcha"){
+            $("#captchaError").show().html(data.responseJSON.message);
+            grecaptcha.reset()
+        }
         else if(data.responseJSON.error == "UserAlreadyExist"){
             $("#emailError").show().html(data.responseJSON.message);
```

Comment:
I think we will need to add grecaptcha.reset() for other cases as well - UserAlreadyExists or InternalError to ensure that the same captcha token is not used in further attempt.

I saw that you catch that in server by maintaining the cache of response token. This minor change will save a round trip time and also less cumbersome from usability perspective.